core/src/main/java/fj/data/List.java

```diff
@@ -1840,4 +1839,31 @@ private void copy() {
     @Override public String toString() {
         return Show.listShow( Show.<A>anyShow() ).show( this ).foldLeft((s, c) -> s + c, "" );
     }
+
+    /**
+     * True if the list is a singleton.  Faster than checking <code>length() == 1</code> for any
+     * list with <code>length() > 1</code>.
+     */
+	public boolean isSingle() {
+		return isNotEmpty() && tail().isEmpty();
+	}
```

Comment:
Ok, so lets update the javadoc for isSingle and remove the compare method and I can merge the PR.